gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -2266,16 +2397,35 @@ bool GazeboRosApiPlugin::spawnAndConform(TiXmlDocument &gazebo_model_xml, std::s
   // todo: should wait for response response_sub_, check to see that if _msg->response == "nonexistant"
 
   gazebo::physics::ModelPtr model = world_->GetModel(model_name);
-  if (model)
+  gazebo::physics::LightPtr light = world_->Light(model_name);
+  if ((isLight && light != NULL) || (model != NULL))
```

Comment:
I don't need to catch that particular case, what is being done here is:
- if the entity to spawn is a light, make sure there's no light with the same name
- if it's a model, then check there's no model with the same name
all the other cases should be fine